src/lib.rs

```diff
@@ -1,27 +1,50 @@
-use std::collections::HashMap;
-
 use borsh::schema::{BorshSchemaContainer, Declaration, Definition, Fields, VariantName};
 use schemars::schema::{RootSchema, Schema};
-use serde::{Deserialize, Serialize};
+use semver::Version;
+use serde::{de, Deserialize, Deserializer, Serialize};
+use std::collections::HashMap;
 
 #[doc(hidden)]
 #[cfg(feature = "__chunked-entries")]
 #[path = "private.rs"]
 pub mod __private;
 
 /// Current version of the ABI schema format.
-pub const SCHEMA_VERSION: &str = env!("CARGO_PKG_VERSION");
+pub const SCHEMA_VERSION_MAJOR: u64 = 0;
+pub const SCHEMA_VERSION_MINOR: u64 = 1;
+pub const SCHEMA_VERSION: &str = "0.1.0";
```

Comment:
Two questions:
- Why would we need to expose both of these, shouldn't one or the other be sufficient? What is the use case for this being viewable externally?
- Why does the `SCHEMA_VERSION` include a patch version if we are only using `major.minor`?

Reply:
> Why does the `SCHEMA_VERSION` include a patch version if we are only using `major.minor`?

Perhaps because `semver` doesn't parse `0.1` as a valid version. - https://docs.rs/semver/latest/semver/struct.Version.html#errors

But then, looking at the code again, @itegulov, we might not actually need `semver` at all.

Reply:
> Why does the SCHEMA_VERSION include a patch version if we are only using major.minor?

I still need to include the patch version inside the `abi.json` (in the future it will also point to metaschema via something like `"$schema": "https://schema.near.org/abi/0.1.0"`). So `SCHEMA_VERSION` is needed to create `AbiRoot`/`ChunkedAbiEntry` with the current schema. Unless you are proposing to drop the patch version from `schema_version` field completely?

> Why would we need to expose both of these, shouldn't one or the other be sufficient? What is the use case for this being viewable externally?

This is tied to the above. I guess we don't necessarily need to expose `MAJOR`/`MINOR` versions, just `SCHEMA_VERSION` should be enough for users. Do you agree?

Reply:
> But then, looking at the code again, @itegulov, we might not actually need semver at all. (since it's just an ordering check):

Yeah I could drop `semver`, I just wasn't sure to which extent do we want to support semver spec. Manually parsing `^([0-9]+)\.([0-9]+)\.([0-9]+)$` is easy enough, but there are also prerelease and build sections that could *technically* happen. `semver` could also be useful if we decide to specify multiple requirements as I suggested in the other thread (`semver` has `VersionReq` struct which can represent `>=0.2 && <0.4`).

Reply:
> Unless you are proposing to drop the patch version from `schema_version` field completely?

No, that's fine to keep as is then. I assumed that was the point of using only `major.minor`. Were you just referring to those `SCHEMA_VERSION_MAJOR ` and `SCHEMA_VERSION_MINOR` variables exposed from the lib?

I'm just a little confused about the use case of these versions being exposed from the library. Is it to be able to assert within tooling that the versions of the statically linked library matches/is compatible?